File: maximum-product-of-three-numbers.ts
function maximumProduct(nums: number[]): number {
    nums.sort((a,b) => b - a);

    const start = nums[0] * nums[1] * nums[2];
    const end = nums.at(-1)! * nums.at(-2)! * nums[0];

    return Math.max(start, end);
};

console.log(maximumProduct([-100,-98,-1,2,3,4]))

File: maximum-product-of-three-numbers.rs
pub fn maximum_product(mut nums: Vec<i32>) -> i32 {
    nums.sort_unstable();
    nums[nums.len() - 3..]
        .iter()
        .product::<i32>()
        .max(nums.last().unwrap() * nums[0] * nums[1])
}
